darts/dataprocessing/transformers/boxcox.py

```diff
@@ -109,21 +134,16 @@ def ts_fit(series: TimeSeries,
 
     @staticmethod
     def ts_transform(series: TimeSeries, lmbda: Union[Sequence[float], pd.core.series.Series]) -> TimeSeries:
+        vals = BoxCox._reshape_in(series)
+        transformed_vals = np.stack([boxcox(vals[:, i], lmbda=lmbda[i]) for i in range(series.width)], axis=1)
+        return series.with_values(BoxCox._reshape_out(transformed_vals, series.width, series.n_samples))
 
-        def _boxcox_wrapper(col):
-            idx = series.pd_dataframe(copy=False).columns.get_loc(col.name)  # get index from col name
-            return boxcox(col, lmbda[idx])
-
-        return TimeSeries.from_dataframe(series.pd_dataframe(copy=False).apply(_boxcox_wrapper))
 
     @staticmethod
     def ts_inverse_transform(series: TimeSeries, lmbda: Union[Sequence[float], pd.core.series.Series]) -> TimeSeries:
```

Comment:
Out of interest, do you know what the benefit is of having these methods defined as static methods in the parent classes?
If they weren't we could access the class methods (like `BoxCox._reshape_in()` more elegantly.

Reply:
I'm not exactly sure actually, I think the idea was maybe to have a more functional way to write these methods (not having to access the object's state), but I agree it might be cleaner to at some point remove this. I'd leave this for later though.